lib/internal/Magento/Framework/Serialize/Serializer/Json.php

```diff
@@ -32,9 +32,9 @@ public function serialize($data)
      * @inheritDoc
      * @since 100.2.0
      */
-    public function unserialize($string)
+    public function unserialize($string, $assoc = true)
```

Comment:
Such change is not allowed: https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/

Let's first find out whether we need such functionality at all.